utils/middleware/error-handling-middleware.ts

```diff
@@ -3,15 +3,23 @@ import { APIGatewayProxyResult } from "aws-lambda";
 import { serializeError } from "serialize-error";
 import { ValidationErrorResponse } from "../response";
 import { INTERNAL_SERVER_ERROR, REQUEST_BODY_INVALID } from "../errors";
-import { ILambdaEvent } from "../../models/provisioning-jobs";
+import { StepFunctionRequestEvent, RequestBodyType, CspInvocation, CspResponse } from "../../models/provisioning-jobs";
 
-export const errorHandlingMiddleware = (): middy.MiddlewareObj<ILambdaEvent, APIGatewayProxyResult> => {
-  const onError: middy.MiddlewareFn<ILambdaEvent, APIGatewayProxyResult> = async (
-    request
-  ): Promise<ValidationErrorResponse | void> => {
-    const error = serializeError(request.error)!;
+export const errorHandlingMiddleware = (): middy.MiddlewareObj<
+  StepFunctionRequestEvent<RequestBodyType> | CspInvocation,
+  APIGatewayProxyResult | CspResponse | ValidationErrorResponse
+> => {
+  const onError: middy.MiddlewareFn<
+    StepFunctionRequestEvent<RequestBodyType> | CspInvocation,
+    APIGatewayProxyResult | CspResponse | ValidationErrorResponse
+  > = async (request): Promise<ValidationErrorResponse | void> => {
+    const error = serializeError(request.error!);
     const errorMessage = error.message;
 
+    if (error.name === "MockCspApiError") {
+      throw request.error;
+    }
```

Comment:
Why is this special? Why does it get re-thrown?

Reply:
Should something happen prior to the `throw`?

Reply:
I guess my question is more -- why are we `throw`ing? Isn't the goal to _handle_ the errors? Why is this one special to be `throw`n rather than being converted to a returned error type?

Reply:
In that case...
```suggestion
    if (error.name === "MockCspApiError") {
      // force state machine task to retry
      throw request.error;
    }
```